src/HealthChecks.NpgSql/DependencyInjection/NpgSqlHealthCheckBuilderExtensions.cs

```diff
@@ -68,12 +68,12 @@ public static class NpgSqlHealthCheckBuilderExtensions
             {
                 throw new ArgumentNullException(nameof(connectionStringFactory));
             }
-
-            builder.Services.AddSingleton(sp => new NpgSqlHealthCheck(connectionStringFactory(sp), healthQuery, connectionAction));
+            var finalName = name ?? NAME;
+            builder.Services.AddTransient(sp => new NpgSqlHealthCheck(finalName, connectionStringFactory(sp), healthQuery, connectionAction));
 
             return builder.Add(new HealthCheckRegistration(
-                name ?? NAME,
-                sp => sp.GetRequiredService<NpgSqlHealthCheck>(),
+                finalName,
+                sp => sp.GetServices<NpgSqlHealthCheck>().Single(x => x.Name == finalName),
```

Comment:
It will blow up existing code like that one:
```c#
builder.AddNpgSql(constr);
builder.AddNpgSql(constr);
```

Reply:
Yeah, we can keep the old behavior for multiple unnamed checks by doing `.Last()` here.